lib/net.js

```diff
@@ -571,7 +572,11 @@ Socket.prototype._read = function(n) {
 
   if (this.connecting || !this._handle) {
     debug('_read wait for connection');
-    this.once('connect', () => this._read(n));
+    this.once('connect', () => {
+      if (!this.destroyed) {
```

Comment:
`_handle.onread` is already set to a noop after destroy so I don't think this is needed. Also `'connect'` should not be emitted if the socket is destroyed while connecting.

Reply:
Then that's a separate problem... the callback would never get called.

Reply:
Hm, I see. I can remove this. I still kind of like it since it makes it more explicit.

Reply:
It would be code that is impossible to cover with tests and this is already explicit